src/schema_impl/src/schemas_impl.rs

```diff
@@ -370,55 +371,77 @@ impl SchemasInner {
         }
 
         // Compute service revision numbers
-        let mut computed_revisions = HashMap::with_capacity(services.len());
-        for service_meta in &services {
-            check_is_reserved(&service_meta.name)?;
+        let mut computed_revisions = HashMap::with_capacity(registration_requests.len());
+        for proposed_service in &registration_requests {
+            check_is_reserved(&proposed_service.name)?;
 
             let instance_type = InstanceTypeMetadata::from_discovered_metadata(
-                service_meta.instance_type.clone(),
-                &service_meta.methods,
+                proposed_service.instance_type.clone(),
+                &proposed_service.methods,
             );
 
             // For the time being when updating we overwrite existing data
-            let revision = if let Some(service_schemas) = self.services.get(&service_meta.name) {
+            let revision = if let Some(existing_service) = self.services.get(&proposed_service.name)
+            {
                 // Check instance type
-                if service_schemas.instance_type != instance_type {
-                    if allow_overwrite {
+                if existing_service.instance_type != instance_type {
```

Comment:
The instance type internally also checks the method set equality, move this check after the new check of missing methods.